### ISSUE TITLE
GetListItems doesn't include the item's ID

### ISSUE THREAD

**Issue by ottothecow** [jasonrollins/shareplum#9]
If you pull a list by name (such as sp_list.GetListItems('All Items')), your results do not have the ID or any way to access the ID that I could find.

That means if you later want to get the attachments for an item, you have no way to reference it.

My quick and dirty solution was to change

        elif viewname:
            viewfields = self.GetView(viewname)['fields']  ## Might be wrong

to

        elif viewname:
            viewfields = self.GetView(viewname)['fields']+['ID']  ## Might be wrong

which seemed to work in my limited testing (although I can't say there aren't cases that wouldn't break it, and it might be better as an option).

**Comment by jasonrollins** [comment]
This is unfortunately a design decision I had to make.  When you create a SharePoint list, it comes with a default view called "All Items".  It's just the name of the first view you have.  It actually doesn't mean "all of the items in this list".

You can create another view and set it as the default view.  Then, when you create new columns for your list these new columns are usually only added to your default view.  If your default view isn't "All Items", your "All Items" view won't have the new column.  Now you have this confusing view called "All Items" that doesn't contain all of the items.

I've tired to make my library match the experience you get when you use SharePoint from a web browser so it feels more intuitive.  So when you pass "All Items" to the GetListItems method you are requesting the view from your SharePoint list called "All Items" and it will only return the info you can see from your browser when you look at the "All Items" view on your SharePoint site.

There are two ways to solve your problem.  The first way is the way I used to do it myself and found that it's not a great idea in the long run.  You can just create a view that contains all of the columns you want and then pass the name of that view to GetListItems.  In the past, I would create a view called "Data".  Then I would select columns like "ID" and "Modified By" and anything I needed.  Then I could simply pass in GetListItems("Data") and get what I needed.  This is pretty quick and dirty.  You will run into issues in the long run because you accidentally forget what this view is for and delete it, or someone else deletes it.  Then you don't remember what all of the columns where.  It can turn into a mess.

This has all been a long explanation to say just use GetListItems(fields=[]).  Just pass in a list of the exact column names you need and you'll have what you want.  Then you won't be tied down to what views are being used for the SharePoint list or how that info is displayed to a browser.

Thanks for the feedback.  I really appreciate it.  I hope you are getting some use out of it.  Let me know if you see any other issues.  I'm sure there is room for improvement.

**Comment by ottothecow** [comment]
Interesting.  I am not very familiar with sharepoint, but I did find it odd that the ID field was missing when it seems like it is a required element for other function calls.

I don't have anything besides read access to the sharepoint site I was looking at, so creating a new view was not an option.  Listing the fields (plus ID) is what I ended up doing.

I suppose a suggestion for improvement might be a convenient way to download files.  For example, once I have pulled a list of attachments, how am I supposed to get one?  I suppose there might be some way to tap into the requests session that is active inside of your module, but I didn't see anything exposed at the user level.  I ended up just authenticating my own parallel requests session and using that to access attachments.
